multiple-erc20-to-erc20/oracle/src/events/processTransfers/index.js

```diff
@@ -19,9 +19,23 @@ const limit = promiseLimit(MAX_CONCURRENT_EVENTS)
 let validatorContract = null
 
 function processTransfersBuilder (config) {
-  return async function processTransfers (transfers, deployedBridges) {
+  return async function processTransfers (transferEvents, deployedBridges) {
     const txToSend = []
-    rootLogger.debug(`Processing ${transfers.length} Transfer events`)
+    rootLogger.debug(`Starting to process ${transferEvents.length} Transfer events`)
+
+    const uniqueTransferEvents = {}
+    transferEvents.forEach(t => {
+      if (uniqueTransferEvents[t.txHash]) {
+        const tx = uniqueTransferEvents[t.txHash]
+        if (tx.txHash === t.txHash && tx.from === t.from && tx.to === t.to && tx.value === t.value && t.data) {
+          uniqueTransferEvents[t.txHash] = t
+        }
+      } else {
+        uniqueTransferEvents[t.txHash] = t
+      }
+    })
```

Comment:
you are using a map with `txHash` as key but later you are checking uniqueness by `txHash` and `from`, `to`, and `value`
what if there where several events in the same tx? let say, that a contract will transfer several assets through the bridge in the same transaction?

Reply:
If there are two events with the same `from`, `to` and `value` than I don't have any way of knowing if they're related or not.